### PR TITLE
system-variables: add description for `tidb_opt_enable_cte_mpp_shared_execution`

### PR DESCRIPTION
**Pull request by winoros**
tidb_opt_enable_mpp_shared_cte_execution

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add the description for the var which controls the CTE's shared scan in MPP.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -2996,6 +2996,19 @@ mysql> desc select count(distinct a) from test.t;
...
+
+> **警告：**
+>
+> 目前 CTE 直接在 TiFlash MPP 环境的执行仍然为实验性特性，代价计算仍在调整，需要搭配变量 `tidb_enforce_mpp` 使用，否则容易出现 CTE 可以在 TiFlash MPP 执行但是依旧是在 TiDB 执行的情况。
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：布尔型
+- 默认值：`OFF`
+- 该变量控制 CTE 是否可以直接在 TiFlash MPP 执行而不是在 TiDB 上执行，打开该开关后，可以极大的提升含大量 CTE 的查询在 TiFlash MPP 的执行速度。
```
Comment (time-and-fate) [inline system-variables.md:3009]:
```suggestion
- 该变量控制 CTE 是否可以直接在 TiFlash MPP 执行而不是在 TiDB 上执行。打开该开关后，可以极大地提升含大量 CTE 的查询在 TiFlash MPP 的执行速度。
```

Comment (time-and-fate) [reply system-variables.md:3009]: 这个理论上除了还是实验功能，应该是没副作用的吧？是的话我没其它问题了。

File: system-variables.md [hunk 2]
```diff
@@ -2996,6 +2996,19 @@ mysql> desc select count(distinct a) from test.t;
 - 这个变量用来控制是否启用 [TiFlash 延迟物化](/tiflash/tiflash-late-materialization.md)功能。注意在 TiFlash [Fast Scan 模式](/tiflash/use-fastscan.md)下，延迟物化功能暂不可用。
 - 当设置该变量为 `OFF` 关闭 TiFlash 延迟物化功能时，如果 `SELECT` 语句中包含过滤条件（`WHERE` 子句），TiFlash 会先扫描查询所需列的全部数据后再进行过滤。当设置该变量为 `ON` 开启 TiFlash 延迟物化功能时，TiFlash 会先扫描下推到 TableScan 算子的过滤条件相关的列数据，过滤得到符合条件的行后，再扫描这些行的其他列数据，继续后续计算，从而减少 IO 扫描和数据处理的计算量。
 
+### `tidb_opt_enable_mpp_shared_cte_execution <span class="version-mark">从 v7.2.0 版本开始引入</span>
+
+> **警告：**
+>
+> 目前 CTE 直接在 TiFlash MPP 环境的执行仍然为实验性特性，代价计算仍在调整，需要搭配变量 `tidb_enforce_mpp` 使用，否则容易出现 CTE 可以在 TiFlash MPP 执行但是依旧是在 TiDB 执行的情况。
```
Comment (qiancai) [inline system-variables.md:3003]:
```suggestion
> 目前 CTE 直接在 TiFlash MPP 环境的执行为实验特性，代价计算仍在调整。如需开启此变量，需要同时开启变量 [`tidb_enforce_mpp`](/system-variables.md#tidb_enforce_mpp-从-v51-版本开始引入)，避免 CTE 可以在 TiFlash MPP 执行的情况下依旧在 TiDB 执行。
```

**Review by qiancai — APPROVED** [review]
/lgtm

File: system-variables.md [hunk 3]
```diff
@@ -2996,6 +2996,19 @@ mysql> desc select count(distinct a) from test.t;
 - 这个变量用来控制是否启用 [TiFlash 延迟物化](/tiflash/tiflash-late-materialization.md)功能。注意在 TiFlash [Fast Scan 模式](/tiflash/use-fastscan.md)下，延迟物化功能暂不可用。
 - 当设置该变量为 `OFF` 关闭 TiFlash 延迟物化功能时，如果 `SELECT` 语句中包含过滤条件（`WHERE` 子句），TiFlash 会先扫描查询所需列的全部数据后再进行过滤。当设置该变量为 `ON` 开启 TiFlash 延迟物化功能时，TiFlash 会先扫描下推到 TableScan 算子的过滤条件相关的列数据，过滤得到符合条件的行后，再扫描这些行的其他列数据，继续后续计算，从而减少 IO 扫描和数据处理的计算量。
 
+### `tidb_opt_enable_mpp_shared_cte_execution <span class="version-mark">从 v7.2.0 版本开始引入</span>
+
+> **警告：**
+>
+> 目前 CTE 直接在 TiFlash MPP 环境执行为实验特性，代价计算仍在调整，不建议在生产环境中未经测试直接使用。如需开启此变量，需要同时开启变量 [`tidb_enforce_mpp`](/system-variables.md#tidb_enforce_mpp-从-v51-版本开始引入)，避免 CTE 可以在 TiFlash MPP 执行的情况下依旧在 TiDB 执行。
```
Comment (qiancai) [inline system-variables.md:3003]:
```suggestion
> 目前公共表表达式 (CTE) 直接在 TiFlash MPP 环境执行为实验特性，代价计算仍在调整，不建议在生产环境中未经测试直接使用。如需开启此变量，需要同时开启变量 [`tidb_enforce_mpp`](/system-variables.md#tidb_enforce_mpp-从-v51-版本开始引入)，避免 CTE 可以在 TiFlash MPP 执行的情况下依旧在 TiDB 执行。
```

File: system-variables.md [hunk 4]
```diff
@@ -2996,6 +2996,19 @@ mysql> desc select count(distinct a) from test.t;
 - 这个变量用来控制是否启用 [TiFlash 延迟物化](/tiflash/tiflash-late-materialization.md)功能。注意在 TiFlash [Fast Scan 模式](/tiflash/use-fastscan.md)下，延迟物化功能暂不可用。
 - 当设置该变量为 `OFF` 关闭 TiFlash 延迟物化功能时，如果 `SELECT` 语句中包含过滤条件（`WHERE` 子句），TiFlash 会先扫描查询所需列的全部数据后再进行过滤。当设置该变量为 `ON` 开启 TiFlash 延迟物化功能时，TiFlash 会先扫描下推到 TableScan 算子的过滤条件相关的列数据，过滤得到符合条件的行后，再扫描这些行的其他列数据，继续后续计算，从而减少 IO 扫描和数据处理的计算量。
 
+### `tidb_opt_enable_mpp_shared_cte_execution <span class="version-mark">从 v7.2.0 版本开始引入</span>
+
+> **警告：**
+>
+> 目前[公共表表达式 (CTE)](/sql-statements/sql-statement-with.md) 直接在 TiFlash MPP 环境执行为实验特性，代价计算仍在调整，不建议在生产环境中未经测试直接使用。如需开启此变量，需要同时开启变量 [`tidb_enforce_mpp`](/system-variables.md#tidb_enforce_mpp-从-v51-版本开始引入)，避免 CTE 可以在 TiFlash MPP 执行的情况下依旧在 TiDB 执行。
```
Comment (qiancai) [inline system-variables.md:3003]:
```suggestion
> 目前[公共表表达式 (CTE)](/sql-statements/sql-statement-with.md) 直接在 TiFlash MPP 环境执行为实验特性，代价计算仍在调整，不建议在生产环境中未经测试直接使用。如需开启此变量，需要同时开启变量 [`tidb_enforce_mpp`](/system-variables.md#tidb_enforce_mpp-从-v51-版本开始引入)，避免 CTE 可以在 TiFlash MPP 执行的情况下依旧在 TiDB 执行。
```

File: system-variables.md [hunk 5]
```diff
@@ -2996,6 +2996,19 @@ mysql> desc select count(distinct a) from test.t;
...
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：布尔型
+- 默认值：`OFF`
+- 该变量控制 CTE 是否可以直接在 TiFlash MPP 执行而不是在 TiDB 上执行。打开该开关后，可以极大地提升含大量 CTE 的查询在 TiFlash MPP 的执行速度。
```
Comment (qiancai) [inline system-variables.md:3009]:
```suggestion
- 该变量控制 CTE 是否可以直接在 TiFlash MPP 执行而不是在 TiDB 上执行。开启该变量后，可以极大地提升含大量 CTE 的查询在 TiFlash MPP 的执行速度。
```

**Review by ran-huang — APPROVED** [review]

File: system-variables.md [hunk 6]
```diff
@@ -2996,6 +2996,19 @@ mysql> desc select count(distinct a) from test.t;
 - 这个变量用来控制是否启用 [TiFlash 延迟物化](/tiflash/tiflash-late-materialization.md)功能。注意在 TiFlash [Fast Scan 模式](/tiflash/use-fastscan.md)下，延迟物化功能暂不可用。
 - 当设置该变量为 `OFF` 关闭 TiFlash 延迟物化功能时，如果 `SELECT` 语句中包含过滤条件（`WHERE` 子句），TiFlash 会先扫描查询所需列的全部数据后再进行过滤。当设置该变量为 `ON` 开启 TiFlash 延迟物化功能时，TiFlash 会先扫描下推到 TableScan 算子的过滤条件相关的列数据，过滤得到符合条件的行后，再扫描这些行的其他列数据，继续后续计算，从而减少 IO 扫描和数据处理的计算量。
 
+### `tidb_opt_enable_mpp_shared_cte_execution` <span class="version-mark">从 v7.2.0 版本开始引入</span>
+
+> **警告：**
+>
+> 目前在 TiFlash MPP 模式下直接执行[公共表表达式 (CTE)](/sql-statements/sql-statement-with.md) 的功能仍在开发中，代价计算仍在调整，不建议在生产环境中未经测试直接使用。如需开启此变量，需要同时开启变量 [`tidb_enforce_mpp`](/system-variables.md#tidb_enforce_mpp-从-v51-版本开始引入)，避免 CTE 可以在 TiFlash MPP 执行的情况下依旧在 TiDB 执行。
```
Comment (qiancai) [inline system-variables.md:3003]:
```suggestion
> 当前版本中该变量控制的功能尚未完全生效，请保留默认值。
```

File: system-variables.md [hunk 7]
```diff
@@ -2996,6 +2996,19 @@ mysql> desc select count(distinct a) from test.t;
...
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：布尔型
+- 默认值：`OFF`
+- 该变量控制 CTE 是否可以直接在 TiFlash MPP 执行而不是在 TiDB 上执行。开启该变量后，可以极大地提升含大量 CTE 的查询在 TiFlash MPP 的执行速度。
```
Comment (qiancai) [inline system-variables.md:3009]:
```suggestion
- 该变量控制非递归的[公共表表达式 (CTE)](/sql-statements/sql-statement-with.md) 是否可以直接在 TiFlash MPP 执行而不是在 TiDB 上执行。
```

File: system-variables.md [hunk 8]
```diff
@@ -2996,6 +2996,19 @@ mysql> desc select count(distinct a) from test.t;
...
+- 该变量目前仅支持控制非递归 CTE 能否在 TiFlash MPP 执行，无法控制递归 CTE。
```
Comment (qiancai) [inline system-variables.md:3010]:
```suggestion
```

**Pull request merged**